### ISSUE TITLE
Review handling of case in UID comparisons

### ISSUE THREAD

**Issue by jonatanschroeder** [PrairieLearn/PrairieLearn#5757]
Seems like some (most? all?) parts of the system treat UID as case-sensitive. In my case I tried adding staff (TAs) to a course, but because the UID had a capital letter it didn't recognize an existing user. Should it be case-sensitive? Maybe this is something to be considered.

**Comment by mwest1066** [comment]
We've always treated UID/UIN as case sensitive (we do exact string comparisons everywhere). Was your situation because some other system was outputting a UID with different case?

**Comment by jonatanschroeder** [comment]
I understand UIN being case sensitive (or at least I don't have a strong case for it), but UID is typically associated to email addresses, which are not case-sensitive. So when I copy-paste a list of "emails" from TAs, and the TA sends a UID starting with upper-case (since it's their first name), I just acted on inertia not considering it could be an issue. In this case it apparently was, and I only noticed it because I was expecting this student to be a previous student and it didn't recognize the name. I "fixed" it by removing the staff member and adding again with lower-case, though I imagine the table of users now has that UID with no association...

**Comment by jonatanschroeder** [comment]
I'm wondering if it may be an interesting idea to have some way to point out, in a query like that done in the staff page, when a UID doesn't have a name/UIN set but there is one that matches ignoring case, to at the very least provide a warning.

**Comment by firasm** [comment]
I think (strongly) that UIDs should be case **in**sensitive because the only time the case sensitivity matters is when TAs or other instructors are providing their emails (UIDs) to be granted access.

Depending on _where_ the email address (UID) is coming from (Canvas, Shib, etc...), it can be in different formats as other systems treat emails as case insensitive.

For a self-hosted instance of PrairieLearn at UBCO, when folks authenticate with CWL, UID is assigned to whatever email they have on file with the university. This is in contrast to what happens on UBC courses on the PL-hosted instance as the UID from UBC is the fake email `cwl@ubc.ca`.

I think this short video demonstrates the friction quite well - for some reason a UBC system stored my colleague's email as `First.Last@ubc.ca` but mine is stored as `first.last@ubc.ca`. Combine this with the autocapitalizing (macOS' fault, not PrairieLearn) and it's chaos.

![pl_UID_unique](https://github.com/PrairieLearn/PrairieLearn/assets/2507459/a3a65e00-c4cb-4f92-b6e8-9cfd86c486f7)

tl;dr: I think UID should be case-insensitive to protect humans from ourselves!
